datafusion/physical-plan/src/coalesce_batches.rs

```diff
@@ -83,6 +83,9 @@ impl CoalesceBatchesExec {
             input.execution_mode(),                 // Execution Mode
         )
     }
+    pub fn set_target_batch_size(&mut self, siz: usize) {
+        self.target_batch_size = siz;
```

Comment:
I think if the `fetch` count is larger than `target_batch_size`, this will introduce some incorrect behavior.

Reply:
I'll set to max()